plugins/command-manager/src/main/java/com/wazuh/commandmanager/jobscheduler/SearchJob.java

```diff
@@ -0,0 +1,224 @@
+                }
+            });
+    }
+
+    public SearchResponse runPitQuery(Client client, String index, Integer resultsPerPage, Object[] searchAfter) throws IllegalStateException {
+        return client.search(
+            pitSearchRequest(client, index, resultsPerPage, searchAfter)
+        ).actionGet(TimeValue.timeValueSeconds(CommandManagerPlugin.SEARCH_QUERY_TIMEOUT));
+    }
+
+    private SearchRequest pitSearchRequest(Client client, String index, Integer resultsPerPage, Object[] searchAfter) {
+        SearchRequest searchRequest = new SearchRequest(index);
+        TermQueryBuilder termQueryBuilder =
+            QueryBuilders.termQuery(Command.COMMAND + "." + Command.STATUS + ".keyword", Status.PENDING);
+        getSearchSourceBuilder()
+            .query(termQueryBuilder)
+            .size(resultsPerPage)
+            .trackTotalHits(true)
+            .pointInTimeBuilder(
+                new PointInTime(client, index).getPointInTimeBuilder()
+            );
+        if( getSearchSourceBuilder().sorts() == null ) {
+            getSearchSourceBuilder()
+                .sort(Command.COMMAND + "." + Command.ORDER_ID + ".keyword", SortOrder.ASC)
+                .sort(Command.COMMAND + "." + Command.TIMEOUT, SortOrder.ASC);
+
+        }
+        if (searchAfter != null) {
+            getSearchSourceBuilder().searchAfter(searchAfter);
+        }
+        searchRequest.source(getSearchSourceBuilder());
+        return searchRequest;
+    }
+
+    public Runnable searchJobRunnable(Client client, String index, Integer resultsPerPage) {
+        return () -> {
+            long consumableHits = 0L;
+            boolean firstPage = true;
+            do {
+                try {
+                    setCurrentPage(
+                        runPitQuery(
+                            client,
+                            index,
+                            resultsPerPage,
+                            getSearchAfter()
+                        )
+                    );
+                    if (firstPage) {
+                        consumableHits = totalHits();
+                        firstPage = false;
+                    }
+                    if ( consumableHits > 0 ) {
+                        handlePage(client, getCurrentPage());
+                        consumableHits -= getPageLength();
+                    }
+                } catch (IOException e) {
+                    log.error("IOException retrieving page: {}", e.getMessage());
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    log.error("ArrayIndexOutOfBoundsException retrieving page: {}", e.getMessage());
+                } catch (IllegalStateException e) {
+                    log.error("IllegalStateException retrieving page: {}", e.getMessage());
+                } catch (RuntimeException e) {
+                    log.error("RuntimeException retrieving page: {}", e.getMessage());
+                    e.printStackTrace();
```

Comment:
Is correct merge this?

Reply:
You mean the `printStackTrace()` call? I just removed it in the latest commit. Good catch.